### PR TITLE
feat(ext/crypto): export elliptic keys as "raw"

### PR DESCRIPTION
**Pull request by diachedelic**
This pull request chips away at the WebCrypto roadmap (#11690). It adds support for the "raw" format when exporting public ECDH/ECDSA keys via the SubtleCrypto.exportKey method.

The WebCrypto specification [explicitly states](https://www.w3.org/TR/WebCryptoAPI/#SubtleCrypto-method-exportKey) that support for the "raw" format is encouraged for interoperability.

As a newbie contributor, are there tests for ext/crypto? I could not find any.

### CODE REVIEW

**Comment by lucacasonato** [comment]
> As a newbie contributor, are there tests for ext/crypto? I could not find any.

There are two types of tests for Web Crypto. Some simple ones in https://github.com/denoland/deno/blob/main/cli/tests/unit/webcrypto_test.ts, and an extensive test suite in the [Web Platform Tests](https://web-platform-tests.org/). You probably just need to update the WPT expectations file to match this new feature. See [Manual: Contributing > Web Platform Tests](https://deno.land/manual/contributing/web_platform_tests).

**Comment by diachedelic** [comment]
I've added some unit tests. Oddly enough, the WPT suite does seem to test exportKey, only importKey.

**Comment by diachedelic** [comment]
I've signed the CLA three times in the last week, and it's still pending.

**Comment by bartlomieju** [comment]
> I've signed the CLA three times in the last week, and it's still pending.

@diachedelic are you sure you used the same GH account to sign the CLA?

**Comment by jamesdiacono** [comment]
Right you are! I had to sign it with my other GitHub account, because the CLAssistant looks at the email address  in the actual commit.

File: ext/crypto/shared.rs [hunk 1]
```diff
@@ -144,7 +144,7 @@ impl RawKeyData {
             )
           })?;
         let point = p384::EncodedPoint::from_bytes(&ec_key.encoded_point)
-          .map_err(|_| data_error("expected valid public EC key"))?;
+          .map_err(|_| data_error("expected valid private EC key"))?;
```
Comment (littledivy) [inline ext/crypto/shared.rs:147]: This is the public encoded point - the error should not be for private key decoding.

**Review by littledivy — APPROVED** [review]
LGTM, just one minor nit.

File: ext/crypto/shared.rs (hunk 1 above)
Comment (diachedelic) [reply ext/crypto/shared.rs]: Oops, fixed.

**Pull request merged**